src/installer/test/HostActivation.Tests/DotNetBuilder.cs

```diff
@@ -65,6 +67,22 @@ public DotNetBuilder AddMicrosoftNETCoreAppFrameworkMockHostPolicy(string versio
             return this;
         }
 
+        /// <summary>
+        /// Use a mock version of HostFxr.
+        /// </summary>
+        public DotNetBuilder UseMockHostFxr()
```

Comment:
This works, but I personally would do it slightly differently:
* call this AddMockHostFxr(string version)
* change the behavior to add a new version folder and into it the mock
* Add the notion of pre-3.0 hostfxr into the naming/parameters somehow
* Add another method to remove all other hostfxr versions (or maybe add one to remove all and then add)

The idea is that we should try to test as close to real world as possible - so in this case, the test would like to test against 2.2 hostfxr, so it should be in 2.2 folder.